adding new access-status-list-element-badge css classes

paulo-graca opened the pull request:
## References
* Fixes #2402

## Description
This PR adds CSS classes to access-status-badge component in order to make possible to differentiate access status based on the value itself.

## Instructions for Reviewers
In order to test it you need to activate in your angular configuration:
``` yaml
# Item Config
item:
  showAccessStatuses: true
```
and for Items and Publications (if you use Entities), inspect the access status value, like: "Open Access" and you should see something like:
``` html
<span class="badge badge-secondary access-status-list-element-badge access-status-open-access-listelement-badge">Open Access</span>
```

List of changes in this PR:
* We just changed the HTML part of the component and replace "." with "-" of the accessStatus value.

artlowel reviewed (changes requested):
Thanks @paulo-graca 

It works, but I just have an inline comment about the performance.

It would also be good to already provide default colors for the access statuses. Note that you can reuse bootstrap's own badge mixin for that e.g.:

```scss
.access-status-open-access-listelement-badge {
  @include badge-variant(map-get($theme-colors, warning));
}
```

this would make the `access-status-open-access-listelement-badge` class idential to the `badge-warning` class (i.e. orange)

paulo-graca commented on src/app/shared/object-collection/shared/badges/access-status-badge/access-status-badge.component.html:
```diff
@@ -1,5 +1,5 @@
 <ng-container *ngIf="showAccessStatus">
   <span *ngIf="accessStatus$ | async as accessStatus">
-    <span class="badge badge-secondary">{{ accessStatus | translate }}</span>
+    <span [class]="'badge badge-secondary access-status-list-element-badge ' + accessStatus.replaceAll('.','-')">{{ accessStatus | translate }}</span>
```
Thank you @artlowel we weren't aware of this re-render issue. We will do as you suggest. Thank you for your feedback!

paulo-graca commented:
@artlowel I think I changed it in the way you suggested. I've also added an empty scss file, but without specifying anything. I think we all don't have a clear idea what access status should look like and you can confirm that on the issue discussion page (https://github.com/DSpace/dspace-angular/issues/2402). 
We are currently working in a custom theme that will address this, when finished, if all agree, we can proposed it to be the default. It will be something like:
![image](https://github.com/DSpace/dspace-angular/assets/877488/f434e4d7-4492-4d23-b5bd-1c4c4ea6b825)
and 
![image](https://github.com/DSpace/dspace-angular/assets/877488/578f6fe2-813d-431f-be71-effe05e24832)

artlowel commented on src/app/shared/object-collection/shared/badges/access-status-badge/access-status-badge.component.ts:
```diff
@@ -57,5 +63,12 @@ export class AccessStatusBadgeComponent {
     );
+
+    // stylesheet based on the access status value
+    this.accessStatus$.pipe(
+      map((accessStatusClass: string) => accessStatusClass.replace(/\./g, '-'))
+    ).subscribe((accessStatusClass: string) => {
+      this.accessStatusClass = accessStatusClass;
+    });
```
Any time you add a subscription, please [store it in a component property](https://github.com/DSpace/dspace-angular/blob/8bf4ba812614e5ed34b2808b79d7e364fd01c2db/src/app/dso-shared/dso-edit-metadata/dso-edit-metadata.component.ts#L122) and then [unsubscribe from it, in the ngOnDestroy method of the component](https://github.com/DSpace/dspace-angular/blob/8bf4ba812614e5ed34b2808b79d7e364fd01c2db/src/app/dso-shared/dso-edit-metadata/dso-edit-metadata.component.ts#L255-L259). That means that we'll unsubscribe whenever that component stops being rendered.

If you don't do that these subscriptions will remain in memory indefinitely, and especially in a case like this, where a new subscription is opened for every search result, that could lead to problems

paulo-graca replied:
I've added the onDestroy to deal with the subscription.

artlowel reviewed (approved):
Thanks @paulo-graca!

artlowel commented:
>We are currently working in a custom theme that will address this, when finished, if all agree, we can proposed it to be the default. It will be something like:
![image](https://github.com/DSpace/dspace-angular/assets/877488/f434e4d7-4492-4d23-b5bd-1c4c4ea6b825)
and 
![image](https://github.com/DSpace/dspace-angular/assets/877488/578f6fe2-813d-431f-be71-effe05e24832)

I like the way that looks
